### PR TITLE
Lesson 11 oop

### PR DESCRIPTION
**Pull request by namelexa**
### Fixed Issues (if relevant)

1. magento/magento2#<issue_number>: Issue title
2. ...

### Manual testing scenarios

1. ...
2. ...

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

File: app/code/AKisilenko/ModuleLesson11/Block/View.php [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+use AKisilenko\ModuleLesson11\Model\FileList;
+use AKisilenko\ModuleLesson11\Model\ReflectionExample;
+
+/**
+ * Class View
+ * @package AKisilenko\ModuleLesson11\Block
+ */
+class View extends \Magento\Framework\View\Element\Template
+{
+    public $constructExampleModel;
```
Comment (chabanenk0) [inline app/code/AKisilenko/ModuleLesson11/Block/View.php]: what is the reason of having this property public? Please share your oppinion. I would recommend making it private

File: app/code/AKisilenko/ModuleLesson11/Model/ReflectionExample.php [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+class ReflectionExample
+{
+
+    const PARENT1 = 'PARENT1';
+    const PARENT2 = 'PARENT2';
+    /**
+     * @return array
+     * @throws \ReflectionException
+     */
+    const MY_CLASS = 'My class';
```
Comment (chabanenk0) [inline app/code/AKisilenko/ModuleLesson11/Model/ReflectionExample.php]: Please move the constant to the top. Annotation above is not related to this constant, it is for the method getConstants

File: app/code/AKisilenko/ModuleLesson11/Model/ReflectionExample.php [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    public function getConstants():array
+    {
+        return (new \ReflectionClass(static::class))->getConstants();
+    }
+
+    /**
+     * @return \ReflectionMethod[]
+     * @throws \ReflectionException
+     */
+    public function getMethods():array
```
Comment (chabanenk0) [inline app/code/AKisilenko/ModuleLesson11/Model/ReflectionExample.php]: please fix codestyle. Add space between the colon and "array":
public function getMethods(): array

File: app/code/AKisilenko/ModuleLesson11/Model/ReflectionExample.php [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    {
+        $item = new  \ReflectionClass(__CLASS__);
+        return $item->getMethods();
+    }
+
```
Comment (chabanenk0) [inline app/code/AKisilenko/ModuleLesson11/Model/ReflectionExample.php:35]: extra newlne,

File: app/code/AKisilenko/ModuleLesson11/Model/ReflectionExample.php [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+}
+
+
```
Comment (chabanenk0) [inline app/code/AKisilenko/ModuleLesson11/Model/ReflectionExample.php]: leave only one new line at the end of the file

File: app/code/AKisilenko/ModuleLesson11/view/frontend/templates/result_view.phtml [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+        <h2>Show info from Di.xml</h2>
+        <?php
+        var_dump($block->getInjectObject());
+        ?>
+    </div>
+
+    <div class="red" style="color: indianred;">
+        <h2>File list </h2>
+        <?php
+        foreach ($block->getFileListObject()->makeFileList() as $item) {
```
Comment (chabanenk0) [inline app/code/AKisilenko/ModuleLesson11/view/frontend/templates/result_view.phtml]: Please try colon variant of foreach (the same approach can be used with other control structures: if, else...
It is commonly used in Magento templates:
<?php foreach ($block->getFileListObject()->makeFileList() as $item): ?>
    <li> <?php echo $item; ?> <?php echo date('F d Y H:i:s.', filemtime($item)); ?>
<?php endforeach; ?>
!!! please check if it works, it's just an example.
Please refer Magento code examples https://github.com/magento/magento2/blob/2.3-develop/app/code/Magento/Directory/view/frontend/templates/currency.phtml#L32

File: app/code/AKisilenko/ModuleLesson11/view/frontend/templates/result_view.phtml [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+            echo "$item" . ' ' . date('F d Y H:i:s.', filemtime($item)) . "\n";
+        }
+        ?>
+    </div>
+
+    <div class="blue" style="color: dodgerblue;">
+        <h2>Reflection Example</h2>
```
Comment (chabanenk0) [inline app/code/AKisilenko/ModuleLesson11/view/frontend/templates/result_view.phtml]: Magento Translations are recommended to be uses while text phrases output

File: app/code/AKisilenko/ModuleLesson9/Block/Widget/Programmatically.php [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+use Magento\Framework\View\Element\Template;
+use Magento\Widget\Block\BlockInterface;
+
+class Programmatically extends Template implements BlockInterface
+{
+
+    /**
+     * @var string
+     */
+
```
Comment (chabanenk0) [inline app/code/AKisilenko/ModuleLesson9/Block/Widget/Programmatically.php:14]: remove extra newline

File: app/code/AKisilenko/ModuleLesson9/view/frontend/templates/widget/homepagebanner.phtml [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+        </h2>
+    <?php endif; ?>
+
+    <?php if ($block->getBannerContent()) : ?>
+        <p class='custom-widget-content'>
+            <?= $block->getBannerContent(); ?>
+        </p>
+    <?php endif; ?>
+<div class="home-banner">
+        <?php echo $block->getLayout()
```
Comment (chabanenk0) [inline app/code/AKisilenko/ModuleLesson9/view/frontend/templates/widget/homepagebanner.phtml:23]: Please create the method in your block AKisilenko\ModuleLesson9\Block\Widget\Programmatically  for CMS blocks rendering and call it here

File: app/code/AKisilenko/ModuleLesson11/Block/View.php (hunk 1 above)
Comment (namelexa) [reply app/code/AKisilenko/ModuleLesson11/Block/View.php]: If my memory serves me, then I had an error and I searched for a reason. I was make it private

File: app/code/AKisilenko/ModuleLesson11/Model/ReflectionExample.php (hunk 2 above)
Comment (namelexa) [reply app/code/AKisilenko/ModuleLesson11/Model/ReflectionExample.php]: done

File: app/code/AKisilenko/ModuleLesson11/Model/ReflectionExample.php (hunk 3 above)
Comment (namelexa) [reply app/code/AKisilenko/ModuleLesson11/Model/ReflectionExample.php]: done

File: app/code/AKisilenko/ModuleLesson11/view/frontend/templates/result_view.phtml (hunk 6 above)
Comment (namelexa) [reply app/code/AKisilenko/ModuleLesson11/view/frontend/templates/result_view.phtml]: done

File: app/code/AKisilenko/ModuleLesson11/view/frontend/templates/result_view.phtml (hunk 7 above)
Comment (namelexa) [reply app/code/AKisilenko/ModuleLesson11/view/frontend/templates/result_view.phtml]: done

File: app/code/AKisilenko/ModuleLesson11/Model/ReflectionExample.php (hunk 5 above)
Comment (namelexa) [reply app/code/AKisilenko/ModuleLesson11/Model/ReflectionExample.php]: done

File: app/code/AKisilenko/ModuleLesson11/Model/FileList.php [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+
+/**
+ * Class MyFiles
+ * @package file_list
+ */
+class FileList
+{
+    public function makeFileList()
+    {
+        $path = realpath('/misc/apps/kalexey/app/code/');
```
Comment (chabanenk0) [inline app/code/AKisilenko/ModuleLesson11/Model/FileList.php]: This directory could be different in other users. 
Please inject \Magento\Framework\Filesystem\DirectoryList $directoryList
to this class and use $path = realpath($this->directoryList->getRoot() . '/app/code/');
to get the proper path

Comment (namelexa) [reply app/code/AKisilenko/ModuleLesson11/Model/FileList.php]: all done

**Pull request merged**